tracer/src/Datadog.Trace/DuckTyping/DuckType.Methods.cs

```diff
@@ -672,14 +689,17 @@ private static class MethodIlHelper
                 if (!targetMethod.IsStatic)
                 {
                     il.Emit(OpCodes.Ldarg_0);
-                    il.Emit(instanceField.FieldType.IsValueType ? OpCodes.Ldflda : OpCodes.Ldfld, instanceField);
+                    if (instanceField is not null)
```

Comment:
This seems like a behavior change. When would `instanceField` be null and as a result we wouldn't try to load the field?

Reply:
This `instanceField` will be null when the `moduleBuilder` is null, because we try to no emit anything when we are doing a `dry run`, in fact, these writes to the `LazyILGenerator` are dropped later. The reason to do it this way is because we want to execute similar branches to the actual run.